great_expectations/datasource/datasource.py

```diff
@@ -196,6 +201,14 @@ def name(self):
     def config(self):
         return copy.deepcopy(self._datasource_config)
 
+    def update_config(self, config: DatasourceConfig) -> None:
+        """
+        See `BaseDatasource.update_config()` - replicated here for legacy parity.
+        """
+        substitution_serializer = DictConfigSerializer(schema=datasourceConfigSchema)
+        raw_config: dict = substitution_serializer.serialize(config)
+        self._datasource_config.update(raw_config)
```

Comment:
What's the best way to do this? Do we perhaps extend the inheritance hierarchy even further so we can place this method in a new base class? `LegacyDatasource` and `BaseDatasource` follow different inheritance hierarchies.